omg/entities/items.py

```diff
@@ -1,12 +1,12 @@
 from abc import ABC, abstractmethod
-from typing import Generic, Iterator, List, TypeVar
+from typing import Iterator, List, Dict, Union
 import arcade
 
-# Define a type variable T
-T = TypeVar("T")
+# Define the type for ItemType
+ItemType = Dict[str, Union[str, float]]
```

Comment:
Can you explain why you changed this generic type `T` to some specialization `Dict[str, Union[str,float]]`?

I thought the same effect about type-hinting would be achieved if you changed `SkillManager(CircularBuffer[ProjectileFactory]):` to `ElementManager(CircularBuffer[Dict[str, Union[str,float]]])`?

Reply:
I did not know that I can do it :D

Reply:
That `Generic[T]` part as the second inherited class was allowing you to insert any type for `T`, for different classes. So, previous `SkillManager` did not have any methods , but it was just specifying `T` to be `ProjectileFactory` before. Now, you only need to change it to the new `Dict[str,Union[str,float]]`